src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -172,14 +183,62 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
                 Console.WriteLine(Strings.ListPkg_AutoReferenceDescription);
             }
 
-            // If any deprecated packages were found, a line is printed
-            // explaining what (D) means.
-            if (deprecatedFound)
+            // If any deprecated packages were found as part of the --outdated command,
+            // a line is printed explaining what (D) means.
+            if (listPackageArgs.IncludeOutdated && deprecatedFound)
             {
                 Console.WriteLine(Strings.ListPkg_DeprecatedPkgDescription);
             }
         }
 
+        private static bool FilterOutdatedPackages(IEnumerable<FrameworkPackages> packages)
+        {
+            return FilterPackages(
+                packages,
+                TopLevelPackagesFilterForOutdated,
+                TransitivePackagesFilterForOutdated);
+        }
+
+        private static bool FilterDeprecatedPackages(IEnumerable<FrameworkPackages> packages)
+        {
+            return FilterPackages(
+                packages,
+                TopLevelPackagesFilterForDeprecated,
+                TransitivePackagesFilterForDeprecated);
+        }
+
+        /// <summary>
+        /// Filters top-level and transitive packages.
+        /// </summary>
+        /// <param name="packages">The <see cref="FrameworkPackages"/> to filter.</param>
+        /// <param name="topLevelPackagesFilter">The filter to be applied on all <see cref="FrameworkPackages.TopLevelPackages"/>.</param>
+        /// <param name="transitivePackagesFilter">The filter to be applied on all <see cref="FrameworkPackages.TransitivePackages"/>.</param>
+        /// <returns><c>True</c> whether at least 1 top-level packages is part of the filtered resultset; otherwise <c>False</c>.</returns>
+        private static bool FilterPackages(
+            IEnumerable<FrameworkPackages> packages,
+            Func<InstalledPackageReference, bool> topLevelPackagesFilter,
+            Func<InstalledPackageReference, bool> transitivePackagesFilter)
+        {
+            var hasTopLevelPackage = false;
+
+            // Filter the packages for deprecated
+            foreach (var frameworkPackages in packages)
+            {
+                frameworkPackages.TopLevelPackages = frameworkPackages.TopLevelPackages
+                    .Where(topLevelPackagesFilter);
+
+                frameworkPackages.TransitivePackages = frameworkPackages.TransitivePackages
+                    .Where(transitivePackagesFilter);
+
+                if (frameworkPackages.TopLevelPackages.Any())
+                {
+                    hasTopLevelPackage = true;
```

Comment:
Another option is to just run a filter and not return and then outside just do something like:
```cs
packages.Any(e => e.TopLevelPackages.Any());
```

Don't want to nitpick though, so you can keep it as is :)

Reply:
Fixed in https://github.com/NuGet/NuGet.Client/pull/2959/commits/985d9ae8e71b9ba21db62a98f53acbf5e51b64e8